### PR TITLE
Fix/add golint fix lint errors

### PR DESCRIPTION
**Pull request by wongni**
*Issue #, if available:* N/A

*Description of changes:* https://goreportcard.com/report/github.com/aws/cluster-api-provider-cloudstack#golint reported lint errors. It seems goreportcard uses golint as linter. Added revive to the enabled linters in golangci settings because revive is a new name of golint, however, revive doesn't report golint warnings related to missing comments for exported consts and/or functions.

*Testing performed:*
'make test' passed
'deploy-app' e2e test passed.
All e2e tests passed. 
```
Ran 13 of 14 Specs in 5540.073 seconds
SUCCESS! -- 13 Passed | 0 Failed | 0 Pending | 1 Skipped
PASS

```
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by maxdrib — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -109,7 +109,7 @@ vet: ## Run go vet on the whole project.
 	go vet ./...
 
 .PHONY: lint
-lint: bin/golangci-lint ## Run linting for the project.
+lint: bin/golangci-lint generate-mocks ## Run linting for the project.
```
Comment (pmotyka) [inline Makefile:112]: Will this result in the generate-mocks target being invoked twice when the test target is called and others that call lint?

Comment (rejoshed) [reply Makefile:112]: It should not. The targets for that are setup well enough that they won't regenerate if they're not older than their inputs.

**Review by rejoshed — APPROVED** [review]

**Comment by rejoshed** [comment]
/lgtm 

You'll need to manually merge though due to the failing e2e test.

**Pull request merged**